test/unit/strainer_unit_test.rb

```diff
@@ -145,4 +145,13 @@ def test_global_filter_clears_cache
     Strainer.global_filter(LateAddedFilter)
     assert_equal 'filtered', Strainer.create(nil).invoke('late_added_filter', 'input')
   end
+
+  def test_add_filter_returns_nil_when_a_filter_module_was_added
+    a = Module.new
+    strainer = Context.new.strainer
+    result1 = strainer.class.add_filter(a)
+    result2 = strainer.class.add_filter(a)
```

Comment:
@dylanahsmith  Yes, you are right. I just happened to notice the bug while reading the code.  Your test case is more clearer than mine and your comment is great. I'm going to close this PR then make a PR with your test case. How about that?